File: core/src/main/java/ca/hojat/zbird/ui/SimpleButton.java
package ca.hojat.zbird.ui;


import com.badlogic.gdx.graphics.g2d.SpriteBatch;
import com.badlogic.gdx.graphics.g2d.TextureRegion;
import com.badlogic.gdx.math.Rectangle;
import ca.hojat.zbird.zbhelpers.AssetLoader;

public class SimpleButton {
    private final float x;
    private final float y;
    private final float width;
    private final float height;

    private final TextureRegion buttonUp;
    private final TextureRegion buttonDown;

    private final Rectangle bounds;

    private boolean isPressed = false;

    public SimpleButton(float x, float y, float width, float height,
                        TextureRegion buttonUp, TextureRegion buttonDown) {
        this.x = x;
        this.y = y;
        this.width = width;
        this.height = height;
        this.buttonUp = buttonUp;
        this.buttonDown = buttonDown;

        bounds = new Rectangle(x, y, width, height);

    }

    public void draw(SpriteBatch batcher) {
        if (isPressed) {
            batcher.draw(buttonDown, x, y, width, height);
        } else {
            batcher.draw(buttonUp, x, y, width, height);
        }
    }

    public void isTouchDown(int screenX, int screenY) {

        if (bounds.contains(screenX, screenY)) {
            isPressed = true;
        }

    }

    public boolean isTouchUp(int screenX, int screenY) {

        // It only counts as a touchUp if the button is in a pressed state.
        if (bounds.contains(screenX, screenY) && isPressed) {
            isPressed = false;
            AssetLoader.flap.play();
            return true;
        }

        // Whenever a finger is released, we will cancel any presses.
        isPressed = false;
        return false;
    }
}

File: core/src/main/java/ca/hojat/zbird/ui/SimpleButton.kt
package ca.hojat.zbird.ui

import ca.hojat.zbird.zbhelpers.AssetLoader
import com.badlogic.gdx.graphics.g2d.SpriteBatch
import com.badlogic.gdx.graphics.g2d.TextureRegion
import com.badlogic.gdx.math.Rectangle

class SimpleButton(
    private val x: Float,
    private val y: Float,
    val width: Float,
    val height: Float,
    private val buttonUp: TextureRegion,
    private val buttonDown: TextureRegion,
) {
    private val bounds = Rectangle(x, y, width, height)
    private var isPressed = false

    fun draw(batcher: SpriteBatch) {
        if (isPressed) {
            batcher.draw(buttonDown, x, y, width, height)
        } else {
            batcher.draw(buttonUp, x, y, width, height)
        }
    }

    fun isTouchDown(screenX: Int, screenY: Int) {
        if (bounds.contains(screenX.toFloat(), screenY.toFloat())) {
            isPressed = true
        }
    }

    fun isTouchUp(screenX: Int, screenY: Int): Boolean {
        // It only counts as a touchUp if the button is in a pressed state.
        if (bounds.contains(screenX.toFloat(), screenY.toFloat()) && isPressed) {
            isPressed = false
            AssetLoader.flappingSound.play()
            return true
        }

        // Whenever a finger is released, we will cancel any presses.
        isPressed = false
        return false
    }
}